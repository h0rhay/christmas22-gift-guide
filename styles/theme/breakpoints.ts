export const REM = 16;

export const dimensionsInPx = {
  small: 960,
  large: 961,
};

export const breakpoints = {
  small: `@media (max-width: ${dimensionsInPx.small / REM}rem)`,
  large: `@media (min-width: ${dimensionsInPx.large / REM}rem)`,
};

const SLIDER_TABLET = 1080;
const SLIDER_MOBILE = 576;

export const tablet = `@media only screen 
and (min-width: ${SLIDER_MOBILE}px) 
and (max-width: ${SLIDER_TABLET}px) `;

export const tabletAndMobile = `@media (max-width: ${SLIDER_TABLET}px) `;
export const mobile = `@media (max-width: ${SLIDER_MOBILE}px) `;
