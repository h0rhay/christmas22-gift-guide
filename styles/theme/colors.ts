import { PrimitiveColorsType } from "types";

const colors: PrimitiveColorsType = {
  red: "rgba(229,58,52,1);",
  bubblegum: "rgba(242,188,200,1);",
  olive: "rgba(87,93,43,1);",
  yolk: "rgba(255,194,92,1);",
  sfYellow: "rgba(255,226,85,1);",
  milk: "rgba(254,251,240,1);",
  milkTransparent: "rgba(254,251,240,0.8);",
  whiteTransparent: "rgba(255, 255, 255, 0.8);",
  black: "rgba(33, 33, 33, 1)",
  white: "rgba(255, 255, 255, 1)",
  grey: {
    main: "rgba(184,184,184,1);",
    mid: "rgba(234,234,234,1);",
    light: "rgba(246,246,246,1);",
  },
};

export default colors;
