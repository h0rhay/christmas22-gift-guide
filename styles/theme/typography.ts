import {
  FontFamilyType,
  FontSizeType,
  FontStyleType,
  FontWeightType,
  TypographyType,
} from "types";
import { CommonSizesType, LineHeight } from "../../types/commonTypes";

const fontFamily: FontFamilyType = {
  formula: '"Formula-Condensed",sans-serif',
  apoc: "Apoc-Revelations",
  universalSansDisplay: "Universal-Sans",
  avalonDemi: "Avalon-Demi",
  default: "-apple-system, BlinkMacSystemFont, Helvetica, Arial, sans-serif",
};

const fontWeight: FontWeightType = {
  light: 300,
  regular: 400,
  medium: 500,
  bold: 700,
};

const fontWeightEntryItems = ["light", "regular", "medium", "bold"] as const;
export type fontWeightEntriesType = typeof fontWeightEntryItems[number];

const fontSize: FontSizeType = {
  small: "1rem",
  medium: "1.25rem",
  large: "1.5rem",
  xlarge: "2rem",
};

const headingFontSize: CommonSizesType = {
  // small: "clamp(1.424rem, 1.75vw, 1.75rem)",
  small: "clamp(1.3rem, calc(1.3rem + ((1vw - 7.68px) * 0.6944)), 1.75rem)",
  medium: "clamp(2rem, calc(2rem + ((1vw - 7.68px) * 0.3472)), 2.25rem);",
  large: "clamp(2.3rem, calc(2.3rem + ((1vw - 7.68px) * 1.3889)), 4rem)",
  xlarge: "clamp(4rem, calc(4rem + ((1vw - 7.68px) * 2.7778)), 6rem)",
};

const lineHeight: LineHeight = {
  heading: "1.1",
  small: "1.2",
  medium: "1.5",
  large: "1.75",
};

const letterSpacing: CommonSizesType = {
  small: "1px",
  medium: "2px",
  large: "3px",
};

const fontStyle: FontStyleType = {
  normal: "normal",
  italic: "italic",
};

const fontStyleEntryItems = ["normal", "italic"] as const;
export type fontStyleEntriesType = typeof fontStyleEntryItems[number];

const space: CommonSizesType = {
  tinyGap: "0.5rem",
  extraSmallGap: "2rem",
  smallGap: "2.5rem",
  mediumGap: "4rem",
  largeGap: "5rem",
  extraSmall: ".5rem",
  small: "1rem",
  medium: "1.5rem",
  large: "2rem",
};

const typography: TypographyType = {
  fontFamily,
  fontWeight,
  fontSize,
  headingFontSize,
  lineHeight,
  letterSpacing,
  fontStyle,
  space,
};

export default typography;
