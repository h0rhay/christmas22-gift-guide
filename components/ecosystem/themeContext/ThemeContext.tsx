import React, { ReactChild, ReactNode } from "react";
import styled, { ThemeProvider } from "styled-components";
import theme from "styles/theme";

const ColoredSurface = styled.div`
  background-color: ${({ theme }) => theme.currentColorGroup.surface};
  * {
    color: ${({ theme }) => theme.currentColorGroup.interface};
  }
`;

type ThemeContextProps = {
  colorGroup?: string;
  children: ReactChild | Array<ReactChild> | ReactNode;
};

const ThemeContext = ({
  children,
  colorGroup = theme.fallbackColorGroup,
}: ThemeContextProps) => {
  const currentColorGroup =
    theme.colorsGroup[colorGroup] ||
    theme.colorsGroup[theme.fallbackColorGroup];
  return (
    <ThemeProvider theme={{ ...theme, currentColorGroup }}>
      <ColoredSurface className="colored-surface">{children}</ColoredSurface>
    </ThemeProvider>
  );
};

export default ThemeContext;
