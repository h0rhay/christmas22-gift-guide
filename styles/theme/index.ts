import { breakpoints } from "./breakpoints";
import typography from "./typography";
import milkColorGroup from "./colorGroups/milk";
import bubblegumColorGroup from "./colorGroups/bubblegum";
import yolkColorGroup from "./colorGroups/yolk";
import oliveColorGroup from "./colorGroups/olive";
import heroColorGroup from "./colorGroups/hero";
import darkColorGroup from "./colorGroups/dark";
import lightColorGroup from "./colorGroups/light";
import colors from "styles/theme/colors";
import { TypographyType, PrimitiveColorsType, ColorsGroupType } from "types";
import { ColorGroupType } from "types/colorsTypes";

const colorsGroup = {
  hero: heroColorGroup,
  milk: milkColorGroup,
  olive: oliveColorGroup,
  bubblegum: bubblegumColorGroup,
  yolk: yolkColorGroup,
  dark: darkColorGroup,
  light: lightColorGroup,
};

const colorsInputs = [
  "hero",
  "milk",
  "bubblegum",
  "yolk",
  "olive",
  "dark",
  "light",
] as const;
export type ColorGroupInputType = typeof colorsInputs[number];

export type ThemeType = {
  typography: TypographyType;
  breakpoints: any; // TODO
  colors: PrimitiveColorsType;
  colorsGroup: ColorsGroupType;
  colorGroupNames: ColorGroupInputType[];
  fallbackColorGroup: ColorGroupInputType;
  currentColorGroup: ColorGroupType;
};

const theme: ThemeType = {
  typography,
  breakpoints,
  colors,
  colorsGroup,
  colorGroupNames: Object.keys(colorsGroup) as ColorGroupInputType[],
  fallbackColorGroup: "milk", //Consider Object.keys(colorsGroup)[0];
  currentColorGroup: heroColorGroup, // dynamic changes
};

export default theme;
