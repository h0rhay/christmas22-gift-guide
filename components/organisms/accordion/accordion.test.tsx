import { render, cleanup } from "@testing-library/react";
import Accordion from "./index";
import { ThemeProvider } from "styled-components";
import Theme from "__mocks__/mockTheme";
import ThemeContext from "components/ecosystem/themeContext/ThemeContext";
import theme, { ColorGroupInputType } from "styles/theme";

const categoriesList = [
  {
    id: 0,
    slug: "beauty",
    title: "Recipe for a sustainable christmas",
    type: "Beauty",
    tasteMaker: "By Artist Name",
    theme: "milk" as ColorGroupInputType,
  },
  {
    id: 1,
    slug: "food",
    title: "Food page",
    type: "Food",
    tasteMaker: "Foobar",
    theme: "olive" as ColorGroupInputType,
  },
];

jest.mock("next/router", () => ({
  useRouter() {
    return {
      route: "/",
      pathname: "",
      query: "",
      asPath: "",
      events: {
        on: jest.fn,
        off: jest.fn,
      },
    };
  },
}));

const renderComponent = ({ colorGroupName = theme.colorGroupNames[0] }) =>
  render(
    <ThemeProvider theme={Theme}>
      <ThemeContext colorGroup={colorGroupName}>
        <Accordion categories={categoriesList}>
          <h1>This is an expected title</h1>
          <p>This is an expected description</p>
        </Accordion>
      </ThemeContext>
    </ThemeProvider>
  );

describe("Accordion test", () => {
  afterEach(cleanup);
  it("should match snapshot", () => {
    const { container } = renderComponent({});
    expect(container).toMatchSnapshot();
  });

  it("Accordion should have the same elements of categories list size", () => {
    const { container } = renderComponent({});
    const accordionTabs = container.getElementsByClassName(
      "xmas-styled-accordion"
    );
    expect(accordionTabs.length).toBe(categoriesList.length);
  });

  it("Accordion should have the same elements of categories list size", () => {
    const { container } = renderComponent({});
    const accordionTabs = container.getElementsByClassName(
      "xmas-styled-accordion"
    );
    expect(accordionTabs.length).toBe(categoriesList.length);
  });

  it("Accordion title, subtitle and accordion header color should match with categoriesList.theme", () => {
    const { container } = renderComponent({});
    const categoryAccordionBackground =
      theme.colorsGroup[categoriesList[0].theme];
    const accordionTab = container.getElementsByClassName(
      "xmas-styled-accordion"
    )[0];
    const title = accordionTab?.getElementsByClassName("accordion-title")[0];
    const subtitle =
      accordionTab?.getElementsByClassName("accordion-subtitle")[0];
    const accordionHeaderTitle = accordionTab?.getElementsByClassName(
      "accordion-header__title"
    )[0];
    expect(title).toHaveStyle(
      `color: ${categoryAccordionBackground?.interface}`
    );
    expect(subtitle).toHaveStyle(
      `color: ${categoryAccordionBackground?.interface}`
    );
    expect(accordionHeaderTitle).toHaveStyle(
      `color: ${categoryAccordionBackground?.interface}`
    );
  });

  it("Accordion odd should have different font family and font weight from accordion even", () => {
    const { container } = renderComponent({});
    const accordionEvenTab = container.getElementsByClassName(
      "xmas-styled-accordion"
    )[0];
    const accordionOddTab = container.getElementsByClassName(
      "xmas-styled-accordion"
    )[1];
    const titleEvenAccordion =
      accordionEvenTab?.getElementsByClassName("accordion-title")[0];
    const titleOddAccordion =
      accordionOddTab?.getElementsByClassName("accordion-title")[0];

    expect(titleEvenAccordion).toHaveStyle(
      `font-family: ${theme.typography.fontFamily.apoc}`
    );
    expect(titleEvenAccordion).toHaveStyle(
      `font-weight: ${theme.typography.fontWeight.light}`
    );
    expect(titleOddAccordion).toHaveStyle(
      `font-family: ${theme.typography.fontFamily.formula}`
    );
    expect(titleOddAccordion).toHaveStyle(
      `font-weight: ${theme.typography.fontWeight.bold}`
    );
  });
});
