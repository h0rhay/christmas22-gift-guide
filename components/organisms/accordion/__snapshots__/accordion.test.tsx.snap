// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion test should match snapshot 1`] = `
<div>
  <div
    class="ThemeContext__ColoredSurface-sc-7o1m9v-0 cumPXi colored-surface"
  >
    <div
      class="ThemeContext__ColoredSurface-sc-7o1m9v-0 fAuFr colored-surface"
    >
      <article
        class="accordion__StyledAccordion-sc-1fl6fpj-0 bJhxsY xmas-styled-accordion"
      >
        <section
          class="siteWidthWrapper__SiteWidthContainer-sc-1kx0urc-0 ckJkzr site_width_wrapper"
        >
          <header>
            <div
              class="accordionHeader__StyledAccordionHeader-sc-r8bbxx-0 kyXyzS"
            >
              <svg
                class="--left-shape"
                fill="none"
                height="32"
                viewBox="0 0 64 32"
                width="64"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  class="accordion-header__path"
                  d="M11.5295 0H0V32H37.4525L64 32C48.8641 32 37.6617 22.8055 31.1436 12.6639C26.7072 5.76137 19.7347 0 11.5295 0Z"
                />
              </svg>
              <div
                class="accordionHeader__CenterShapeContainer-sc-r8bbxx-1 YmNsT"
              >
                <h4
                  class="accordion-header__title"
                >
                  Beauty
                </h4>
              </div>
              <svg
                class="--right-shape"
                fill="none"
                height="32"
                viewBox="0 0 64 32"
                width="64"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  class="accordion-header__path"
                  d="M11.5295 0H0V32H37.4525L64 32C48.8641 32 37.6617 22.8055 31.1436 12.6639C26.7072 5.76137 19.7347 0 11.5295 0Z"
                />
              </svg>
            </div>
            <h2
              class="title__StyledTitle-sc-z5nezq-0 lfjBkS xmas-title accordion-title"
            >
              Recipe for a sustainable christmas
            </h2>
            <h3
              class="subtitle__StyledSubTitle-sc-1awsyh-0 hrOwwR xmas-subtitle accordion-subtitle"
            >
              By Artist Name
            </h3>
          </header>
          <div
            class="accordion-bottom-space"
          />
        </section>
      </article>
    </div>
    <div
      class="ThemeContext__ColoredSurface-sc-7o1m9v-0 rMDNb colored-surface"
    >
      <article
        class="accordion__StyledAccordion-sc-1fl6fpj-0 dOTFgk xmas-styled-accordion"
      >
        <section
          class="siteWidthWrapper__SiteWidthContainer-sc-1kx0urc-0 ckJkzr site_width_wrapper"
        >
          <header>
            <div
              class="accordionHeader__StyledAccordionHeader-sc-r8bbxx-0 bPHbRf"
            >
              <svg
                class="--left-shape"
                fill="none"
                height="32"
                viewBox="0 0 64 32"
                width="64"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  class="accordion-header__path"
                  d="M11.5295 0H0V32H37.4525L64 32C48.8641 32 37.6617 22.8055 31.1436 12.6639C26.7072 5.76137 19.7347 0 11.5295 0Z"
                />
              </svg>
              <div
                class="accordionHeader__CenterShapeContainer-sc-r8bbxx-1 isLeGf"
              >
                <h4
                  class="accordion-header__title"
                >
                  Food
                </h4>
              </div>
              <svg
                class="--right-shape"
                fill="none"
                height="32"
                viewBox="0 0 64 32"
                width="64"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  class="accordion-header__path"
                  d="M11.5295 0H0V32H37.4525L64 32C48.8641 32 37.6617 22.8055 31.1436 12.6639C26.7072 5.76137 19.7347 0 11.5295 0Z"
                />
              </svg>
            </div>
            <h2
              class="title__StyledTitle-sc-z5nezq-0 lfjBkS xmas-title accordion-title"
            >
              Food page
            </h2>
            <h3
              class="subtitle__StyledSubTitle-sc-1awsyh-0 bhAuAQ xmas-subtitle accordion-subtitle"
            >
              Foobar
            </h3>
          </header>
          <div
            class="accordion-bottom-space"
          />
        </section>
      </article>
    </div>
  </div>
</div>
`;
